src/main/io/ledstrip.c

```diff
@@ -535,6 +537,10 @@ static void applyLedWarningLayer(bool updateNow, uint32_t *timer)
                 warningFlags |= 1 << WARNING_FAILSAFE;
             if (!ARMING_FLAG(ARMED) && !ARMING_FLAG(OK_TO_ARM))
                 warningFlags |= 1 << WARNING_ARMING_DISABLED;
+#ifdef MAG
+            if (masterConfig.mag_hardware != MAG_NONE && !compassIsWorking())
```

Comment:
I would make it a generic function i.e. `isHardwareFailing` so we can actually see if any of the hardware is not functioning properly (i.e. GPS is configured but not detected etc)